fennel/testing/execute_aggregation.py

```diff
@@ -337,6 +339,28 @@ def get_val(self) -> List:
         return list(self.counter.keys())
 
 
+class QuantileState(AggState):
+    def __init__(self, default, p):
+        self.p = p
+        self.default = default
+        self.vals = []
+
+    def add_val_to_state(self, val):
+        self.vals.append(val)
+        return self.get_val()
+
+    def del_val_from_state(self, val):
+        self.vals.remove(val)
+        return self.get_val()
+
+    def get_val(self) -> List:
+        if len(self.vals) == 0:
+            return self.default
+        self.vals.sort()
+        index = int(math.floor(self.p * len(self.vals))) + 1
+        return self.vals[min(len(self.vals) - 1, max(index - 1, 0))]
```

Comment:
This would sort all the data for every read ?
Can we use a sorted tree data structure instead, inserts are log(n) and reads are log(n) too. 

from sortedcontainers import SortedList
class QuantileState(AggState):
    def __init__(self, default, p):
        self.p = p
        self.default = default
        self.vals = SortedList()

    def add_val_to_state(self, val):
        self.vals.add(val)
        return self.get_val()

    def del_val_from_state(self, val):
        self.vals.remove(val)
        return self.get_val()

    def get_val(self):
        if len(self.vals) == 0:
            return self.default
        index = int(math.floor(self.p * (len(self.vals) - 1)))
        return self.vals[index]